s3_storage_provider.py

```diff
@@ -65,25 +70,32 @@ def __init__(self, hs, config):
         if "secret_access_key" in config:
             self.api_kwargs["aws_secret_access_key"] = config["secret_access_key"]
 
+        self._download_pool = ThreadPool(name="s3-download-pool", maxthreads=40)
```

Comment:
I think we could benefit from making pool size a config option, if it's relatively easy.

Reply:
done in 2628892